cpp/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -2171,6 +2171,24 @@ class ArrowDeserializer {
     return Status::OK();
   }
 
+  Status Visit(const ExtensionType& type) {
+    auto storage_type = type.storage_type();
+
+    ArrayVector out_chunks(data_->num_chunks());
+    for (int i = 0; i < data_->num_chunks(); i++) {
+      auto chunk = data_->chunk(i);
+      auto ext_data = chunk->data()->Copy();
+      ext_data->type = storage_type;
+      auto storage_result = MakeArray(ext_data);
```

Comment:
Perhaps you can just write `auto storage_result = checked_cast<const ExtensionArray&>(*chunk).storage();`?

Reply:
Thanks, that's indeed quite a bit simpler!